contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,161 @@
+# VueJs
+
+Author: [Lu Lechuan](https://github.com/LuLechuan)
+
+VueJs (also known as Vue) is an open-source progressive [JavaScript framework](https://en.wikipedia.org/wiki/JavaScript_framework) for building user interfaces. It is designed to improve code quality and maintainablity.
+
+## Creating a Simple Project in VueJs
+
+Printing the VueJs version of "Hello World" (a minimal example to show how easy it is to integrate VueJs into your web project):
+
+```HTML
+<!--This is the main HTML file-->
+<body>
+  <div id="root">
+    <h2>{{ message }}</h2>
+  </div>
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+</body>
+```
+
+```js
+// This is inside the javacript file
+new Vue ({
+  el: '#root',
+
+  data: {
+    message: "Hello World"
+  }
+});
+```
+
+Step by step guide:
+
+1. Import VueJs CDN and the JavaScript file in the main HTML file.
+```HTML
+  <script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+  <script src="the_path_to_the_javacript_file.js"></script>
+```
+
+2. Create a Vue instance in the JavaScript file; bind the instance to one of the component in our html file (the `root` element).
+  - In this case, only the `root` component can be accessed in VueJs while the rest are unaffected. This is how we progressively plug in VueJs into our projects.
+```js
+  new Vue ({
+    el: '#root',
+  });
+```
+```HTML
+  <div id="root"></div>
+```
+
+3. Specify our data (message: "Hello World") in the Vue instance.
+```js
+  data: {
+    message: "Hello World"
+  }
+```
+
+5. Pass the message to the HTML file using double curly brackets.
```

Comment:
5 after 3?

Reply:
Alternatively, you can do this:
>**Step 4**: some text
>```html
>some text
>```

(to avoid code block being less indented than the numbered item)
![image](https://user-images.githubusercontent.com/1673303/38880431-4d3a9cc6-4298-11e8-8d2c-2e310f9b6cfc.png)